common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -809,4 +811,21 @@ private boolean cachedAccessTokenKidMatchesKeystoreKid(@Nullable final String at
 
         return deviceKid.equals(atKid);
     }
+
+    /**
+     * Using this method to load the {@link OpenIdProviderConfiguration}
+     */
+    public void loadOpenIdProviderConfiguration() {
```

Comment:
Wouldn't it be better for this method to actually return the configuration? and then the caller uses the configuration

Reply:
In this use case, the caller (`CIAMAuthority`) doesn't need to access the configuration. I wanted to avoid the confusion of devs thinking they would need to utilize the Open ID Configuration themselves by simply not returning it, as it used internally by `MicrosoftStsAuthorizationRequest`

Reply:
Okay, but why is the caller meant to invoke this method? If caller needs to invoke this i.e. `CIAMAuthority` then perhaps the method should move to `CIAMAuthority` and become a private method there?

Reply:
I tried to set it up in a way that any class that uses `MicrosoftStsOauth2Strategy` can decide whether or not to load this configuration. For example, if you want to change AAD Authorities to use OpenID for the authorization endpoint, they would just have to call this method in `MicrosoftStsOauth2Strategy`, without worrying about supplying the authority url as that will be provided from `MicrosoftStsOauth2Strategy`.

Reply:
Rework this to pass a boolean parameter in the constructor to opt into openidconfig